### PR TITLE
Add eslint plugin

### PR DESCRIPTION
**Pull request by MattPutnam**
At this point I'm looking mainly for feedback on the plugin itself. This is both a PR on its own merits, and the beginning of a proposal for a new linting rule for Best Practices.

### CODE REVIEW

**Comment by chris-pardy** [comment]
@MattPutnam plugin looks good, I think there's an easy way we can add roll-up/typescript to make this 💯

**Review by suzil — APPROVED** [review]
like it

**Comment by ThomasHayOwnUp** [comment]
What's the logic behind making this its own repo instead of adding it to best-practices?

**Comment by MattPutnam** [comment]
@ThomasHayOwnUp Two things:

1. ESLint's architecture doesn't like working with locally-defined plugins. It seems the main way to do it is through yet another plugin.
2. This seems like a useful plugin outside of our own best-practices. I tried to find an existing plugin and only found one, and it didn't have any configurability.

File: package.json [hunk 1]
```diff
@@ -2,18 +2,46 @@
...
   "author": "Own Up <dev+eslint@ownup.com>",
   "license": "MIT",
   "scripts": {
-    "format": "prettier --write '**/*.js'",
-    "lint": "eslint . --ext js",
-    "test": "mocha test/use-numeric-separator.test.js"
+    "build": "rollup --config rollup.config.js && tsc --build tsconfig.build.json",
+    "type-check": "tsc --noEmit",
+    "format": "prettier --write '**/*.ts'",
+    "lint": "tslint --project .",
```
Comment (chris-pardy) [inline package.json:18]: use `eslint`

File: README.md [hunk 2]
```diff
@@ -1 +1,60 @@
...
+0xDEADBEEF
+0b10101010
+
+// Correct:
+1000 // short number
+10_000_000
+100.0000001 // decimals are not split
+0xDE_AD_BE_EF
+0b1010_1010
+6.02214e+23
```
Comment (ThomasHayOwnUp) [inline README.md:27]: Nit: Consider adding a comment about what this example is meant to illustrate. Also consider putting all the examples that are not from the `Incorrect` section at the end of the `Correct` section, not in the middle or moving them to their own section.

**Review by ThomasHayOwnUp — APPROVED** [review]
Code looks good, please make the switch to eslint that Chris suggested

**Comment by chris-pardy** [comment]
@MattPutnam should be all set now

File: README.md (hunk 2 above)
Comment (MattPutnam) [reply README.md]: 👍

**Pull request merged**